wordpress-mobile/WordPress-Android
# Fixes #8002: Add ability to change username

Fixes #8002: Add ability to change username

To test:
- Open Account Settings.
- Tap "username".
- Enter base name and wait for suggestions.
- Pick a suggestion.
- Enter confirmation.
- Check the username was updated.
- Hit back, check the username is updated on the "Me" screen.

Note: this is the merge of the feature branch. Code and feature in this branch was already reviewed and tested in:
- https://github.com/wordpress-mobile/WordPress-Android/pull/10409
- https://github.com/wordpress-mobile/WordPress-Android/pull/10414 
- https://github.com/wordpress-mobile/WordPress-Android/pull/10436

There was one remaining issue and it was fixed in FluxC via https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1378 - This PR targets a recent FluxC beta (1.4.0-beta-1) including that fix.

cc @jd-alexander @malinajirka 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. Updated in 8827832
